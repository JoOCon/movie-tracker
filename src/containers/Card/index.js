import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import { addFavorite } from '../../actions';
import { postFavorites } from '../../api/apiCalls';
import './style.css';

export const Card = (props) => {
  const { movie, user, addFavorite } = props;
  const { poster_path, vote_average } = movie;
  const rating = vote_average / 2;

  const handleFavorite = () => {
    const favoriteData = {...movie, user_id: user.id};
    addFavorite(favoriteData);
    postFavorites(favoriteData);
  };

  return (
    <div className='card'>
      <div className='favorite-wrapper'>
        <button 
          className='favorite' 
          onClick={() => handleFavorite()}>
        </button>
        <StarRatings 
          starDimension={'1em'} 
          rating={rating} 
          numberOfStars={5} 
          starRatedColor={'gold'} 
        />
      </div>
      <img className='poster-image' src={poster_path} alt='movie data' />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userData
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  postFavorites: (movie) => dispatch(postFavorites(movie))
});

Card.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object,
  addFavorite: PropTypes.func,
  postFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
