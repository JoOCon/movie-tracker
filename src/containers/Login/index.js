import React, { Component } from 'react';
import './style.css';
import { login } from '../../api/apiCalls'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await login(this.state);
    // this.setState({ userName: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='email' placeholder='email' type='email' value={email} onChange={this.handleChange} />
        <input name='password' placeholder='password' type='password' value={password} onChange={this.handleChange} />
        <button>LogIn</button>
      </form>
    );
  }
}
