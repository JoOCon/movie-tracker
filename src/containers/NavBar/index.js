import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};