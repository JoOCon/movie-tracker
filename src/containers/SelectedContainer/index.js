import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { manageSelectedMovie } from '../../actions';
import './style.css';

export const SelectedContainer = (props) => {
  const { selectedMovie, manageSelectedMovie } = props;
  const {
    overview,
    poster_path,
    release_date,
    title
  } = props.selectedMovie;

  const displayInfo = () => { 
    if (poster_path){
      return (
        <div className='selected-container' onClick={() => manageSelectedMovie(selectedMovie)}>
          <img className='poster-image' src={poster_path} alt='movie data' />
          <div className='selected-movie-content'>
            <h1>{title}</h1>
            <h3>RELEASE: {release_date}</h3>
            <p>DETAILS: {overview}</p>
          </div>
        </div>);
    } else {
      return <div></div>;
    }
  };

  return (
    <div className='selected-wrapper'>{displayInfo()}</div>
  );
};

export const mapStateToProps = (state) => ({
  selectedMovie: state.selectedMovie
});

export const mapDispatchToProps = (dispatch) => ({
  manageSelectedMovie: (movie) => dispatch(manageSelectedMovie(movie))
});

SelectedContainer.propTypes = {
  selectedMovie: PropTypes.object,
  manageSelectedMovie: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(SelectedContainer);
