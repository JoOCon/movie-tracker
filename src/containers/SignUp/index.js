import React, { Component } from 'react';
import './style.css';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({[name]: value});
  }

  handleSubmit = (e) => {
    //this is where we hook up our action for submiting a post to the backend.
    this.setState({userName: '', email: '', password: ''});
  }

  render() {
    const {userName, email, password} = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='userName' value={userName} onChange={this.handleChange}/>
        <input name='email' value={email} onChange={this.handleChange}/>
        <input name='password' value={password} onChange={this.handleChange}/>
        <button>SignUp</button>
      </form>
    );
  }
}
