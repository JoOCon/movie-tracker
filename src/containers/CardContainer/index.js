import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import './style.css';

const CardContainer = (props) => {
  const displayMovies = props.movies.map(movie => <Card key={movie.id} movie={movie} />);
  return (
    <div className="card-container">
      {displayMovies}
    </div>
  );
};

const mapStateToProps = (state) => ({
  movies: state.moviesData
});

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps)(CardContainer);
