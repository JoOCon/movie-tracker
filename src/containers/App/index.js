import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { addMovies } from '../../actions';
import { fetchMovies } from '../../api/apiCalls';
import CardContainer from '../CardContainer';
import FavoriteContainer from '../FavoriteContainer';
import Login from '../../containers/Login';
import SignUp from '../../containers/SignUp';
import { Header } from '../../components/Header';
import './style.css';

export class App extends Component {
  componentDidMount = () => {
    this.fetchOpeningMovies();
  }

  fetchOpeningMovies = async () => {
    const { addMovies } = this.props;
    const openingMovies = await fetchMovies();
    addMovies(openingMovies);
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Route path='/' component={Header} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/signup' component={SignUp} />
          <Route exact path='/' component={CardContainer} />
          <Route exact path='/:landingPage' component={CardContainer} />
          <Route exact path='/:user/favorites' component={FavoriteContainer} />
        </div>
      </BrowserRouter>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (moviesData) => dispatch(addMovies(moviesData))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default connect(null, mapDispatchToProps)(App);
