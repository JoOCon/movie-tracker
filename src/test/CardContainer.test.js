import React from 'react';
import { CardContainer, mapStateToProps } from '../containers/CardContainer';
import { shallow } from 'enzyme';
import * as mockData from '../test/mockData';

describe('CardContainer tests', () => {
  let wrapper;
  const { mockCleanMoviesDataArray } = mockData;
  beforeEach(() => {
    wrapper = shallow( 
      <CardContainer 
        movies={mockCleanMoviesDataArray}
      /> 
    );
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should have a movies data array in props', () => {
      const { mockCleanMoviesDataArray } = mockData;
      const mockState = {moviesData: mockCleanMoviesDataArray};
      const expected = {movies: mockCleanMoviesDataArray};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });
});
