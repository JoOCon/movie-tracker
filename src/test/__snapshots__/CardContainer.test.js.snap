// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    movies={
        Array [
            Object {
              "movie_id": 383498,
              "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
              "poster_path": "https://image.tmdb.org/t/p/w200/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
              "release_date": "2018-05-15",
              "title": "Deadpool 2",
              "vote_average": 7.5,
            },
            Object {
              "movie_id": 299536,
              "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
              "poster_path": "https://image.tmdb.org/t/p/w200/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
              "release_date": "2018-04-25",
              "title": "Avengers: Infinity War",
              "vote_average": 8.3,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          movie={
                    Object {
                              "movie_id": 383498,
                              "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
                              "poster_path": "https://image.tmdb.org/t/p/w200/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
                              "release_date": "2018-05-15",
                              "title": "Deadpool 2",
                              "vote_average": 7.5,
                            }
          }
/>,
        <Connect(Component)
          movie={
                    Object {
                              "movie_id": 299536,
                              "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
                              "poster_path": "https://image.tmdb.org/t/p/w200/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
                              "release_date": "2018-04-25",
                              "title": "Avengers: Infinity War",
                              "vote_average": 8.3,
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "movie_id": 383498,
            "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
            "poster_path": "https://image.tmdb.org/t/p/w200/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
            "release_date": "2018-05-15",
            "title": "Deadpool 2",
            "vote_average": 7.5,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "movie_id": 299536,
            "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
            "poster_path": "https://image.tmdb.org/t/p/w200/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
            "release_date": "2018-04-25",
            "title": "Avengers: Infinity War",
            "vote_average": 8.3,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            movie={
                        Object {
                                    "movie_id": 383498,
                                    "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
                                    "poster_path": "https://image.tmdb.org/t/p/w200/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
                                    "release_date": "2018-05-15",
                                    "title": "Deadpool 2",
                                    "vote_average": 7.5,
                                  }
            }
/>,
          <Connect(Component)
            movie={
                        Object {
                                    "movie_id": 299536,
                                    "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
                                    "poster_path": "https://image.tmdb.org/t/p/w200/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
                                    "release_date": "2018-04-25",
                                    "title": "Avengers: Infinity War",
                                    "vote_average": 8.3,
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "movie_id": 383498,
              "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
              "poster_path": "https://image.tmdb.org/t/p/w200/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
              "release_date": "2018-05-15",
              "title": "Deadpool 2",
              "vote_average": 7.5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "movie_id": 299536,
              "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
              "poster_path": "https://image.tmdb.org/t/p/w200/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
              "release_date": "2018-04-25",
              "title": "Avengers: Infinity War",
              "vote_average": 8.3,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
